### PR TITLE
Deprecate get_communicator() in favor of get_mpi_communicator().

### PR DESCRIPTION
**Pull request by drwells**
We use the later name more often than the former.

### CODE REVIEW

File: include/deal.II/grid/tria.h [hunk 1]
```diff
@@ -1810,11 +1810,21 @@ class Triangulation : public Subscriptor
...
+
+  /**
+   * Return the MPI communicator used by this triangulation. In the case of
+   * a serial Triangulation object, MPI_COMM_SELF is returned.
+   *
+   * @deprecated Use get_mpi_communicator() instead.
+   */
+  MPI_Comm DEAL_II_DEPRECATED_EARLY_WITH_COMMENT(
+    "Access the MPI communicator with get_mpi_communicator() instead.")
+    get_communicator() const;
```
Comment (Rombur) [inline include/deal.II/grid/tria.h:1827]: You forgot `get_communicator()`

File: include/deal.II/dofs/dof_handler.h [hunk 2]
```diff
@@ -1218,7 +1218,16 @@ class DoFHandler : public Subscriptor
...
+  get_mpi_communicator() const;
+
+  /**
+   * Return MPI communicator used by the underlying triangulation.
+   *
+   * @deprecated Use get_mpi_communicator() instead.
+   */
+  MPI_Comm DEAL_II_DEPRECATED_EARLY_WITH_COMMENT(
+    "Access the MPI communicator with get_mpi_communicator() instead.")
+    get_communicator() const;
```
Comment (Rombur) [inline include/deal.II/dofs/dof_handler.h:1230]: You forgot `get_communicator()`

File: include/deal.II/distributed/tria_base.h [hunk 3]
```diff
@@ -97,7 +97,7 @@ namespace parallel
      * Return MPI communicator used by this triangulation.
      */
     virtual MPI_Comm
-    get_communicator() const override;
+    get_mpi_communicator() const override;
```
Comment (Rombur) [inline include/deal.II/distributed/tria_base.h:100]: You need to add the deprecated function back.

**Review by Rombur — CHANGES_REQUESTED** [review]

File: include/deal.II/distributed/tria_base.h (hunk 3 above)
Comment (drwells) [reply include/deal.II/distributed/tria_base.h:100]: Since this inherits from `Triangulation` we don't need to add a shim here - just in the base class.

**Review by Rombur — APPROVED** [review]

**Pull request merged**